packages/protocol/contracts/libs/LibAddress.sol

```diff
@@ -34,11 +45,12 @@ library LibAddress {
         view
         returns (bool result)
     {
-        try IERC165Upgradeable(addr).supportsInterface(interfaceId) returns (
-            bool _result
-        ) {
-            result = _result;
-        } catch { }
+        if (AddressUpgradeable.isContract(addr)) {
```

Comment:
if (!AddressUpgradeable.isContract(addr))  return false;

Reply:
> if (!AddressUpgradeable.isContract(addr)) return false;

It returns false if not contract.

Reply:
Yeah, just a trick to reduce the level of {}

Reply:
No worries.